pkg/applier/applier_test.go

```diff
@@ -499,11 +499,15 @@ func TestProcessApplyEvent(t *testing.T) {
 		isDestroy: false,
 	}
 
-	err := eh.processApplyEvent(ctx, formApplyEvent(event.ApplyFailed, deploymentObj, fmt.Errorf("test error")).ApplyEvent, s.ApplyEvent, objStatusMap, unknownTypeResources)
-	expectedError := ErrorForResource(fmt.Errorf("test error"), idFrom(deploymentID))
+	resourceMap := make(map[core.ID]client.Object)
+	resourceMap[idFrom(deploymentID)] = deploymentObj
+
+	err := eh.processApplyEvent(ctx, formApplyEvent(event.ApplyFailed, deploymentObj, fmt.Errorf("test error")).ApplyEvent, s.ApplyEvent, objStatusMap, unknownTypeResources, resourceMap)
+	expectedError := ErrorForResourceWithResource(fmt.Errorf("test error"), idFrom(deploymentID), deploymentObj)
 	testutil.AssertEqual(t, expectedError, err, "expected processPruneEvent to error on apply %s", event.ApplyFailed)
+	testutil.AssertEqual(t, expectedError.Body(), err.Body())
```

Comment:
I think this assertion may be more useful if we compare the output of `ToCSE` to an explicitly instantiated struct.

For example:
```golang
expectedCSE := v1beta1.ConfigSyncError{
...
}
testutil.AssertEqual(t, expectedCSE, err.ToCSE())
```

Reply:
Done